st2common/st2common/models/api/action.py

```diff
@@ -497,6 +497,27 @@ class ActionAliasAPI(BaseAPI, APIUIDMixin):
                 "type": "array",
                 "items": {"type": "string"},
                 "description": "Possible parameter format."
+            },
+            "aliases": {
```

Comment:
We do support regex in formats. The whole point is not exposing this in `!help` and (in the future) other public lists. Specifying long regexes in formats,—or five formats for the same command,—is cluttering the list, it's not user-friendly and I'm pretty sure it hurts adoption. We could somehow flag the formats to not show up in help, but I couldn't come up with a better idea. 

I do share your concern though, I just think it's the best option in our current state: it's fully reusing the current code, it doesn't require you to learn anything new, and it's completely backwards compatible (just don't use it if you don't have to).